### PR TITLE
WIP global: add open search support

### PR DESCRIPTION
**Pull request by lnielsen**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -25,14 +26,43 @@
 
 
 @pytest.fixture()
-def app():
+def app_config():
+    if 'OPENSEARCH_CLIENT_URI' not in os.environ:
+        return {}
+
+    params = {}
```
Comment (lnielsen) [inline tests/conftest.py:33]: To be aligned with docker-services-cli

File: invenio_search/compat.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#
+# This file is part of Invenio.
+# Copyright (C) 2021 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Compatibility layer for abstracting away Elasticsearch and OpenSearch."""
+
+import pkg_resources
```
Comment (lnielsen) [inline invenio_search/compat.py:11]: Use ``importlib_metadata`` instead.

File: invenio_search/utils.py [hunk 3]
```diff
@@ -76,7 +75,7 @@ def build_index_name(index, prefix=None, suffix=None, app=None):
     :param suffix: The suffix to append to the index name.
     :param app: Flask app passed to ``prefix_index`` and ``suffix_index``.
     """
-    if not isinstance(index, six.string_types):
+    if not isinstance(index, str):
         index = build_index_from_parts(*index)
     index = prefix_index(index, prefix=prefix, app=app)
     index = suffix_index(index, suffix=suffix, app=app)
```
Comment (lnielsen) [inline invenio_search/utils.py:81]: ES_VERSION checks can likely be removed due to Elasticsearch v6 no longer being maintained.

**Comment by ntarocco** [comment]
Superseded by https://github.com/inveniosoftware/invenio-search/pull/216

**Pull request closed without merging**